fizz-buzz-streams/src/test/java/com/pubsap/eng/streams/FizzBuzzAllTopologySpec.java

```diff
@@ -0,0 +1,192 @@
+        SpecificAvroSerde<Output> outputSerde = new SpecificAvroSerde<>(mockedRegistry);
+
+        itemSerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), false);
+        inputSerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), false);
+        outputSerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), false);
+        inputKeySerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), true);
+        outputKeySerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), true);
+
+        mockedRegistry.register(inputTopic + "-value", Input.SCHEMA$);
+        mockedRegistry.register(outputTopic + "-value", Output.SCHEMA$);
+
+        TimeWindows windows = TimeWindows
+
+                .of(Duration.ofSeconds(20))
+
+                .advanceBy(Duration.ofSeconds(20));
+
+        Topology topology = Main.buildTopology(
+                config,
+                windows,
+                itemSerde,
+                inputSerde,
+                inputKeySerde,
+                outputSerde,
+                outputKeySerde
+        );
+
+        testDriver = new TopologyTestDriver(topology, properties);
+
+        inputTopic = testDriver.createInputTopic(inputTopicName, inputKeySerde.serializer(), inputSerde.serializer());
+        outputTopic = testDriver.createOutputTopic(outputTopicName, outputKeySerde.deserializer(), outputSerde.deserializer());
+
+    }
+
+    @After
+    public void closeTestDriver(){
+        testDriver.close();
+    }
+
+    @Test
+    public void allTopologyTestSamePlayerSameWindow() {
+
+        //Given
+        final List<KeyValue<InputKey,Input>> inputValues = Arrays.asList(
+                new KeyValue<>(new InputKey("player-1"),new Input(3,Instant.parse("2020-02-14T14:26:00Z"))),
+                new KeyValue<>(new InputKey("player-1"),new Input(5,Instant.parse("2020-02-14T14:26:01Z"))),
+                new KeyValue<>(new InputKey("player-1"),new Input(15,Instant.parse("2020-02-14T14:26:02Z"))),
+                new KeyValue<>(new InputKey("player-1"),new Input(3,Instant.parse("2020-02-14T14:26:03Z"))),
+                new KeyValue<>(new InputKey("player-1"),new Input(5,Instant.parse("2020-02-14T14:26:04Z"))),
+                new KeyValue<>(new InputKey("player-1"),new Input(15,Instant.parse("2020-02-14T14:26:05Z")))
+        );
+
+        //When
+        inputTopic.pipeKeyValueList(inputValues);
+
+        //Then
+        final Map<OutputKey, Output> expectedWordCounts = mkMap(
+                mkEntry(new OutputKey("player-1","2020-02-14T14:26:00Z","2020-02-14T14:26:20Z"),new Output(2,2,2))
+        );
+        assertThat(outputTopic.readKeyValuesToMap(),equalTo(expectedWordCounts));
+        assertTrue(outputTopic.isEmpty());
+    }
+
+    @Test
+    public void allTopologyTestDifferentPlayersSameWindow() {
```

Comment:
Suggestion:  `topologyShouldGroupEventsFromSameClient()`
BTW: let's forget the idea of _"player"_ (this is a random thing I wrote) and describe the Sapients as "client" so (`client-1`, `client-2`)